File: applykit.py
#!/usr/bin/env python3
"""applykit CLI — deterministic outreach draft from JD + CV blurb."""
from __future__ import annotations

import argparse
import re
from pathlib import Path

SKILLS = [
    "laravel", "php", "vue", "nuxt", "react", "next", "flutter", "dart",
    "node", "python", "django", "docker", "redis", "mysql", "postgres",
    "nginx", "chapa", "api", "rest", "typescript", "javascript",
]


def matched(jd: str) -> list[str]:
    toks = set(re.findall(r"[a-z][a-z0-9+.#-]*", jd.lower()))
    return [s for s in SKILLS if s in toks or any(s in t for t in toks)]


def build(name: str, title: str, cv: str, jd: str) -> str:
    m = matched(jd)
    match_line = (
        f"Overlap with this role: {', '.join(m[:8])}."
        if m
        else "I work across Laravel, Vue, Flutter, and Node in production."
    )
    return "\n".join(
        [
            f"Hi — I'm {name}, {title} based in Addis Ababa.",
            "I saw your opening and wanted to reach out.",
            cv.strip(),
            match_line,
            "Happy to share a CV or jump on a short call. Thanks for reading.",
        ]
    )


def main() -> None:
    p = argparse.ArgumentParser(description="Generate a job outreach draft")
    p.add_argument("--jd", required=True, help="Path to job description text")
    p.add_argument("--cv", required=True, help="Path to CV blurb text")
    p.add_argument("--name", default="Biruk Endrias")
    p.add_argument("--title", default="Senior Full Stack Developer")
    args = p.parse_args()
    print(
        build(
            args.name,
            args.title,
            Path(args.cv).read_text(encoding="utf-8"),
            Path(args.jd).read_text(encoding="utf-8"),
        )
    )


if __name__ == "__main__":
    main()

File: applykit.js
const SKILLS = [
  "laravel", "php", "vue", "nuxt", "react", "next", "flutter", "dart",
  "node", "python", "django", "docker", "redis", "mysql", "postgres",
  "nginx", "chapa", "api", "rest", "graphql", "typescript", "javascript",
];

function tokens(text) {
  return (text || "").toLowerCase().match(/[a-z][a-z0-9+.#-]*/g) || [];
}

function matchedSkills(jd) {
  const set = new Set(tokens(jd));
  return SKILLS.filter((s) => set.has(s) || [...set].some((t) => t.includes(s)));
}

function draft({ name, title, cv, jd }) {
  const matched = matchedSkills(jd);
  const matchLine = matched.length
    ? `Overlap with this role: ${matched.slice(0, 8).join(", ")}.`
    : "I work across Laravel, Vue, Flutter, and Node in production.";
  const roleHint = (jd.match(/(?:looking for|seeking|hiring)\s+([^.!\n]{8,60})/i) || [])[1];
  const opener = roleHint
    ? `I saw the opening for ${roleHint.trim()}.`
    : "I saw your opening and wanted to reach out.";

  return [
    `Hi — I'm ${name}, ${title} based in Addis Ababa.`,
    opener,
    cv.trim(),
    matchLine,
    "Happy to share a CV or jump on a short call. Thanks for reading.",
    "",
    "— Matched bullets to emphasize —",
    ...matched.slice(0, 6).map((s) => `• Emphasize recent ${s} work and metrics from production.`),
  ].join("\n");
}

document.getElementById("run").addEventListener("click", () => {
  const payload = {
    name: document.getElementById("name").value.trim() || "Biruk Endrias",
    title: document.getElementById("title").value.trim() || "Full Stack Developer",
    cv: document.getElementById("cv").value,
    jd: document.getElementById("jd").value,
  };
  document.getElementById("out").value = draft(payload);
});

document.getElementById("copy").addEventListener("click", async () => {
  const out = document.getElementById("out").value;
  if (out) await navigator.clipboard.writeText(out);
});
